### PR TITLE
Register Datadog classloader as parallel capable

### PR DESCRIPTION
**Pull request by realark**
- Register Datadog classloader as parallel capable and add a test to assert instance is not locked on during classloading

### CODE REVIEW

File: dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/DatadogClassLoader.java [hunk 1]
```diff
@@ -24,6 +28,12 @@ public DatadogClassLoader(URL bootstrapJarLocation, URL agentJarLocation, ClassL
     bootstrapProxy = new BootstrapClassLoaderProxy(new URL[] {bootstrapJarLocation}, null);
   }
 
+  /** Public for testing only */
+  @Override
+  public Object getClassLoadingLock(String className) {
+    return super.getClassLoadingLock(className);
+  }
```
Comment (mar-kolya) [inline dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/DatadogClassLoader.java]: It looks like this method already had enough visibility to not needing this. If you remove this method tests still pass.

File: dd-java-agent/agent-bootstrap/src/test/groovy/datadog/trace/bootstrap/DatadogClassLoaderTest.groovy [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    threadHoldLockPhase.arriveAndAwaitAdvance()
+    synchronized (ddLoader.getClassLoadingLock(className2)) {
+      acquireLockFromMainThreadPhase.arrive()
+    }
+    thread.join()
+    boolean applicationDidNotDeadlock = true
+
+    then:
+    applicationDidNotDeadlock
```
Comment (mar-kolya) [inline dd-java-agent/agent-bootstrap/src/test/groovy/datadog/trace/bootstrap/DatadogClassLoaderTest.groovy]: Failure of this test would result in test runs just hanging forever. Would it be possible to implement this so test actually exits on failure? E.g. using two additional threads and making 'main' thread just wait reasonable amount of time?

File: dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/DatadogClassLoader.java (hunk 1 above)
Comment (realark) [reply dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/DatadogClassLoader.java]: Interesting. Looks like a protected method is visible to spock tests. Pretty useful for avoiding this problem of escalating exposure for testing. Removing this method.

File: dd-java-agent/agent-bootstrap/src/test/groovy/datadog/trace/bootstrap/DatadogClassLoaderTest.groovy (hunk 2 above)
Comment (realark) [reply dd-java-agent/agent-bootstrap/src/test/groovy/datadog/trace/bootstrap/DatadogClassLoaderTest.groovy]: 👍 I added another thread and used spock's timeout annotation to fail this test after 1 minute.

**Pull request merged**